app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt

```diff
@@ -70,6 +80,37 @@ class ShareFragment : AppCompatDialogFragment() {
         setStyle(STYLE_NO_TITLE, R.style.ShareDialogStyle)
     }
 
+    private val networkCallback = object : ConnectivityManager.NetworkCallback() {
+        override fun onLost(network: Network?) {
+            reloadDevices()
+        }
+
+        override fun onAvailable(network: Network?) {
+            lifecycleScope.launch {
+                delay(NETWORK_RESTORED_DELAY)
+                reloadDevices()
+            }
+        }
+    }
+
+    private fun reloadDevices() {
+        context?.let {
+            val list = lifecycleScope.async(Dispatchers.IO) {
```

Comment:
I'm not sure I follow why this is done async so it might be helpful to add a comment around it.

Is it for adding the delay to updating the UI, or is it to give the account manager enough time to load new devices? If it's the latter then we may have to do a device refresh instead to make sure we have the most recent devices (i.e. when you come online, the device list may have changed).

Reply:
It's the latter. How do I do a device refresh? I thought I was by building a new device list :/